### PR TITLE
Multiple ZKStateReaders created in the overseer node 

### PR DESCRIPTION
**Pull request by noblepaul**
The ZkStateReader used in the ZkController#cloudManager is different . This leads to different visibility.

### CODE REVIEW

File: solr/solrj/src/java/org/apache/solr/client/solrj/impl/CloudSolrClient.java [hunk 1]
```diff
@@ -1368,6 +1368,14 @@ private static LBHttpSolrClient createLBHttpSolrClient(Builder cloudSolrClientBu
      */
     @Deprecated
     public Builder() {}
+
+
+    /**Provide an already created {@link ClusterStateProvider} instance
```
Comment (claygoddard) [inline solr/solrj/src/java/org/apache/solr/client/solrj/impl/CloudSolrClient.java]: nit: fix comment format

File: solr/solrj/src/java/org/apache/solr/client/solrj/impl/CloudSolrClient.java [hunk 2]
```diff
@@ -1368,6 +1368,14 @@ private static LBHttpSolrClient createLBHttpSolrClient(Builder cloudSolrClientBu
...
+     *
+     */
+    public Builder(ClusterStateProvider stateProvider) {
+      this.stateProvider = stateProvider;
```
Comment (claygoddard) [inline solr/solrj/src/java/org/apache/solr/client/solrj/impl/CloudSolrClient.java]: Does it make more sense to just accept the `ZkStateReader` instead? And then have `build()` create the `ClusterStateProvider` where it checks the different scenarios (state reader vs zk hosts vs http client config).

Comment (noblepaul) [reply solr/solrj/src/java/org/apache/solr/client/solrj/impl/CloudSolrClient.java]: The Builder already accepts a ClusterStateProvider. So, this is OK

Comment (claygoddard) [reply solr/solrj/src/java/org/apache/solr/client/solrj/impl/CloudSolrClient.java]: That's fair. SG.

**Comment by claygoddard** [comment]
LGTM

**Comment by claygoddard** [comment]
Sorry about that, accidentally hit the close button!

**Pull request merged**